src/transformState.js

```diff
@@ -4,8 +4,28 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
   // write code here
-}
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
 
+      case 'removeProperties':
+        for (const names in action.keysToRemove) {
+          delete state[action.keysToRemove[names]];
+        }
+        break;
+
+      case 'clear':
+        for (const states in state) {
+          delete state[states];
+        }
+        break;
+      default:;
```

Comment:
don't leave default statement blank
```suggestion
       default:
        break;
```